bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/TableViewer.java

```diff
@@ -118,8 +122,65 @@ public TableViewer(Composite parent, int style) {
 	public TableViewer(Table table) {
 		this.table = table;
 		hookControl(table);
+		// listen to tree resizes.
+		table.addControlListener(new ControlListener() {
```

Comment:
Shouldn't it be added only when setInncrementalDisplay is invoked with a value > 0 (and removed if value is then set to 0) ?

Reply:
yes.. i will fix.